src/Symfony/Bundle/ArgumentResolver/PayloadArgumentResolver.php

```diff
@@ -73,7 +73,7 @@ private function getExpectedInputClass(Request $request): ?string
 
         $context = $this->serializationContextBuilder->createFromRequest($request, false, RequestAttributesExtractor::extractAttributes($request));
 
-        return $context['input'] ?? $context['resource_class'];
+        return $context['input']['class'] ?? $context['resource_class'];
```

Comment:
```suggestion
        return $context['input']['class'] ?? $context['resource_class'] ?? null;
```

To be 100% sure that it will not crash.

Reply:
Done ! Thanks for your feedback